torch/distributed/_tensor/device_mesh.py

```diff
@@ -241,6 +244,72 @@ def __eq__(self, other: object) -> bool:
             return True
         return self.mesh.equal(other.mesh)
 
+    def _create_child_mesh(self, mesh_dim: int) -> "DeviceMesh":
```

Comment:
instead of putting these logic inside DeviceMesh, I think it's probably cleaner to put this logic to `MeshEnv`, and we can add two APIs in `MeshEnv`:
1. create/slice_child_mesh
2. get_parent_mesh

For DeviceMesh we can just expose `__getitem__` API and inside this API we call mesh_resources respective method

and also we can put a map to map between a child mesh to parent mesh in MeshEnv:
i.e. child_to_parent: Dict[torch.Tensor, DeviceMesh], where key is the mesh tensor and the value is the parent device mesh. In this way we don't need to put the parent mesh inside DeviceMesh (which I think would make DeviceMesh implementation cleaner).